### PR TITLE
[ARCTIC-1695][ARCTIC-1646]: Support hadoop 3.3

### LINKED ISSUE #1646: [Bug]: Build failed when change hadoop version to 3.2.1
### What happened?

Arctic currently relies on Hadoop version 2.9.1. When I tried to connect to HDFS with version 3.2.1, it failed due to configuration parsing errors. After I changed the Hadoop dependency version to 3.2.1, the project failed to build.

### Affects Versions

master

### What engines are you seeing the problem on?

AMS

### How to reproduce

1build project with command:
```sql
 mvn clean pakacge -Dhadoop.version=3.2.1 
```

### Relevant log output

WechatIMG3488

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #1695: [Bug]: Terminal can not work after upgrade spark to 3.3
### What happened?

When I using terminal, I encounter follow exception
```
2023/07/13 10:36:42 java.lang.RuntimeException: run with ugi request failed.
	at com.netease.arctic.table.TableMetaStore.doAsUgi(TableMetaStore.java:370)
	at com.netease.arctic.table.TableMetaStore.lambda$doAs$0(TableMetaStore.java:345)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1873)
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:345)
	at com.netease.arctic.server.terminal.TerminalSessionContext$ExecutionTask.get(TerminalSessionContext.java:193)
	at com.netease.arctic.server.terminal.TerminalSessionContext$ExecutionTask.get(TerminalSessionContext.java:164)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NoSuchFieldError: transportConf
	at org.apache.spark.network.netty.NettyBlockTransferService.init(NettyBlockTransferService.scala:72)
	at org.apache.spark.storage.BlockManager.initialize(BlockManager.scala:507)
	at org.apache.spark.SparkContext.(SparkContext.scala:610)
	at org.apache.spark.SparkContext$.getOrCreate(SparkContext.scala:2714)
	at org.apache.spark.sql.SparkSession$Builder.$anonfun$getOrCreate$2(SparkSession.scala:942)
	at scala.Option.getOrElse(Option.scala:189)
	at org.apache.spark.sql.SparkSession$Builder.getOrCreate(SparkSession.scala:936)
	at com.netease.arctic.server.terminal.local.LocalSessionFactory.lazyInitContext(LocalSessionFactory.java:122)
	at com.netease.arctic.server.terminal.local.LocalSessionFactory.create(LocalSessionFactory.java:58)
	at com.netease.arctic.server.terminal.TerminalSessionContext.lazyLoadSession(TerminalSessionContext.java:158)
	at com.netease.arctic.server.terminal.TerminalSessionContext.access$200(TerminalSessionContext.java:40)
	at com.netease.arctic.server.terminal.TerminalSessionContext$ExecutionTask.lambda$get$0(TerminalSessionContext.java:194)
	at com.netease.arctic.table.TableMetaStore.doAsUgi(TableMetaStore.java:365)
	... 11 more
```

### Affects Versions

0.5.0

### What engines are you seeing the problem on?

AMS

### How to reproduce

_No response_

### Relevant log output

_No response_

### Anything else

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

fix #1695 and fix #1646 

To support with Hadoop 3.3. 

## Brief change log

- Add profile `hadoop2` and `hadoop3`
- Use `spark 3.1` as the terminal backend when choosing profile `hadoop2`, and use `spark 3.3` when choosing `hadoop3`

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [X] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by baiyangtx** [comment]
image
It's Works

image

File: core/src/main/java/com/netease/arctic/io/ArcticHadoopFileIO.java [hunk 1]
```diff
@@ -212,17 +208,13 @@ public Iterable<FileInfo> listPrefix(String prefix) {
...
-      Path toDelete = new Path(prefix);
-      FileSystem fs = getFs(toDelete);
+      Path prefixToDelete = new Path(prefix);
+      FileSystem fs = getFs(prefixToDelete);
       try {
-        if (!fs.delete(toDelete, true)) {
-          throw new IOException("Fail to delete directory:" + prefix + " recursively, " +
-              "file system return false, need to check the hdfs path");
-        }
+        return fs.delete(prefixToDelete, true /* recursive */);
```
Comment (zhoujinsong) [inline core/src/main/java/com/netease/arctic/io/ArcticHadoopFileIO.java:214]: It appears that in Hadoop 3, `FileSystem#delete` no longer returns a boolean value. However, removing the return value check from Hadoop 2 may result in the failure of the delete operation being ignored.

File: pom.xml [hunk 2]
```diff
@@ -622,18 +639,34 @@
...
+                <guava.version>14.0.1</guava.version>
+                <hive.version>2.1.1</hive.version>
+                <hadoop.version>2.9.2</hadoop.version>
+                <terminal.spark.version>3.1.2</terminal.spark.version>
+                <terminal.spark.major.version>3.1</terminal.spark.major.version>
             </properties>
         </profile>
         <profile>
-            <id>hive-v2.1</id>
+            <id>hadoop3</id>
```
Comment (zhaomin1423) [inline pom.xml:658]: In hadoop3, whether to consider using hadoop-client-api and hadoop-client-runtime, they exclude third-party dependencies, we can refer to spark

Comment (baiyangtx) [reply pom.xml:658]: his will make our pom file difficult to be compatible with Hadoop 2/3 in the simplest way, so I will still use a unified dependency approach in this PR.

File: ams/server/pom.xml [hunk 3]
```diff
@@ -55,12 +54,32 @@
             <groupId>com.netease.arctic</groupId>
             <artifactId>arctic-core</artifactId>
             <version>${project.version}</version>
+            <exclusions>
```
Comment (zhoujinsong) [inline ams/server/pom.xml]: Can we handle these exclude logics in the `core` module instead of requiring dependent parties to exclude them?

**Review by zhoujinsong — APPROVED** [review]

**Pull request merged**